### PR TITLE
Add "Media Updates in Chrome 61" /updates article

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Hello @mounirlamouri @dalecurtis,
Let me know what you think of this draft.

Live preview at https://pr-4857-dot-web-central.appspot.com/web/updates/2017/07/chrome-61-media-updates

### CODE REVIEW

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+the video back to windowed mode as you would expect.
+
+<figure>
+  <img src="/web/updates/images/2017/07/auto-fullscreen-rotate.png"
+       alt="Automatic video fullscreen when device is rotating">
+</figure>
+
+This magic behaviour only happens when:
+
+- device is an Android phone
```
Comment (mounirlamouri) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: underline not tablet?

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- user's screen orientation is set to "Auto-rotate"
+- video size is at least 200x200px
+- video use native controls
+- video is currently playing
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
```
Comment (mounirlamouri) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Worth pointing out that if the page has a screen lock, it does not happen too.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
```
Comment (mounirlamouri) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Should you remind developers that they can implement this themselves? I believe you mention this in your best practice article, maybe point to it?

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+snippet that shows you what Chrome is roughly doing behind the scenes.
+
+    var video = document.querySelector('video');
+    var selectedVideoTrackIndex;
+
+    document.addEventListener('visibilitychange', function() {
+      if (document.hidden) {
+        // Disable video track when page is hidden.
+        selectedVideoTrackIndex = video.videoTracks.selectedIndex;
+        video.videoTracks[selectedVideoTrackIndex].selected = false;
```
Comment (mounirlamouri) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: that's behind a flag, right?

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      } else {
+        // Re-enable video track when page is not hidden anymore.
+        video.videoTracks[selectedVideoTrackIndex].selected = true;
+      }
+    });
```
Comment (mounirlamouri) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Should we suggest that developers reduce the quality of the streem if they think it's worth it?

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 1 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 2 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 3 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 5 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 4 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: It is for now yes but this code is meant to explain what is going on. WDYT?

**Comment by beaufortfrancois** [comment]
@mounirlamouri Your feedback has been addressed at https://github.com/google/WebFundamentals/pull/4857/commits/ee832e1fe0878ddd50929772b35d6dd97f674b6e. Thanks!

**Comment by mounirlamouri** [comment]
lgtm :+1:

**Comment by jondahlke** [comment]
LGTM

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: A round up of the media updates in Chrome 61.
+
+{# wf_updated_on: 2017-07-17 #}
+{# wf_published_on: 2017-07-17 #}
+{# wf_tags: news,chrome61,media #}
+{# wf_featured_image: /web/updates/images/generic/animations.png #}
+{# wf_featured_snippet: Background video track optimization and automatic video fullscreen when device is rotating features are there! #}
```
Comment (dalecurtis) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: "Background video track optimizations and rotate-for-fullscreen are here!" ?

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+# Media Updates in Chrome 61 {: .page-title }
+
+{% include "web/_shared/contributors/beaufortfrancois.html" %}
+
+- Chrome now [disables video tracks when a MSE video is played in the
+  background](#background-video-track-optimization) to optimize performance. 
+- A video will [automatically go fullscreen when device is
```
Comment (dalecurtis) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: s/A video/Video/
s/automatically//

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  rotating](#auto-fullscreen-rotate).
```
Comment (dalecurtis) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: s/rotating/rotated/

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## Background video track optimization (MSE only) {: #background-video-track-optimization}
+
+To improve battery life, Chrome now disables video tracks when the video is
+played in the background if this one uses [Media Source Extensions (MSE)].
```
Comment (dalecurtis) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: s/if this one uses/if the video uses/

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+And here are some restrictions:
+
+- This optimization only applies to videos with a [keyframe] distance < 5s.
+- If the video doesn't contain any audio tracks, the video will be
+  automatically paused when played in the background.
+
+[Chromium Bug](https://bugs.chromium.org/p/chromium/issues/detail?id=663999)
+
+## Automatic video fullscreen when device is rotating {: #auto-fullscreen-rotate }
```
Comment (dalecurtis) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Rotate-for-fullscreen ? s/rotating/rotated/ at least

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 7 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 8 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 9 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

**Comment by beaufortfrancois** [comment]
Thank you @dalecurtis! I've addressed your feedback.
I'm still using "Automatic video fullscreen when device is rotated" though as "rotate-for-fullscreen" seems cryptic at first glance in my opinion. If you feel strongly, we may want to revisit my decision.

**Comment by beaufortfrancois** [comment]
@jpmedley You may review this article now. Thanks!

**Comment by beaufortfrancois** [comment]
(gentle ping) @jpmedley

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+- This optimization only applies to videos with a [keyframe] distance < 5s.
+- If the video doesn't contain any audio tracks, the video will be
+  automatically paused when played in the background.
+
+[Chromium Bug](https://bugs.chromium.org/p/chromium/issues/detail?id=663999)
+
+## Automatic video fullscreen when device is rotated {: #auto-fullscreen-rotate }
+
+When a video is playing in the viewport, rotating the device to landscape
+orientation will now fullscreen the video. Rotating the device to portrait puts
```
Comment (jpmedley) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: This sentence is a little clunky. Try:

If you rotate a device to landscape while a video is playing, playback will automatically switch to fullscreen mode.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+the video back to windowed mode as you would expect.
```
Comment (jpmedley) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Delete 'as you would expect.' It doesn't add any information, and I'm not sure that I would expect this.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+Note that you can implement manually this behaviour yourself (see [Mobile Web Video
```
Comment (jpmedley) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Change 'implement manually' to 'manually implement'.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+This magic behaviour only happens when:
+
+- device is an Android phone (not a tablet)
+- user's screen orientation is set to "Auto-rotate"
+- video size is at least 200x200px
+- video uses native controls
+- video is currently playing
+- at least 75% of the video is visible (on-screen)
+- orientation rotates by 90 degrees (not 180 degrees)
+- there is no fullscreen element yet
```
Comment (jpmedley) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: I didn't know what a fullscreen element was. Link to this: https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenElement

Or if you know of something more appropriate, use that.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+- screen is not locked using the Screen Orientation API
```
Comment (jpmedley) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Maybe consider linking to the spec. https://w3c.github.io/screen-orientation/

The API isn't documented on MDN. It looks like it shipped in the last version before we started contributing to MDN :(

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 12 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 11 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 14 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

File: src/content/en/updates/2017/07/chrome-61-media-updates.md (hunk 15 above)
Comment (beaufortfrancois) [inline src/content/en/updates/2017/07/chrome-61-media-updates.md]: Done.

**Comment by beaufortfrancois** [comment]
@jpmedley Thank you! I've addressed your feedback and merged back your changes.

**Pull request merged**